Compare parameters uploaded to and already in BETYdb (#43)
az-digitalag/model-vignettes: BioCro/DARPA/upload_params_data/BETY/Compare_BETY_to_upload.Rmd

---
title: "Compare BETY and upload_params_data"
---
  
  This script compares the results of what is in BETY to what has been uploaded to it, in an effort to identify mistakes or duplicates. 

> Data from BETY is selecte with the following SQL query: select * from traits where treatment_id in (9000000015, 9000000016, 9000000017, 9000000018, 9000000019, 9000000020)

Reviewer: ```suggestion
Data from BETY is selected with the following SQL query: select * from traits where treatment_id in (9000000015, 9000000016, 9000000017, 9000000018, 9000000019, 9000000020)
```

> The results are saved as csv in the upload_params_data/BETY folder as the most recent 'update_at' date. 

Reviewer 2: ```suggestion
The results are saved as csv in the `upload_params_data/BETY` folder, with a name that matches the most recent date in the `updated_at` column. 
```

Load libraries and both sets of data
```{r}
options(scipen = 999)
library(dplyr)
library(arsenal)

> #import newest BETY file

Reviewer 2: Usually when I feel compelled to add comments to code chunks, that's a sign that the code associated with each comment should probably be its own chunk. I'd recommend splitting this code chunk up into three chunks with corresponding text descriptions.

Reviewer 2: And I think it's a common convention for the loading in of libraries to be its own code chunk anyways.

filename <- max(as.numeric(substr(list.files(pattern = "csv"), 1,8)))
in_bety <- read.csv(paste0(filename, ".csv"), header = T)
#import full upload_params_data data file
in_upload <- read.csv("../phys_parameters.csv",  header = T)
```

> Convert the treatment and variable ID columns to match

Reviewer 2: ```suggestion
Convert the treatment and variable ID columns of BETY data to match upload data format
```

                               treatment_id == "9000000019" ~ "outdoor 5 cm density",
                               treatment_id == "9000000020" ~ "outdoor JollyG soil"),
         variable = case_when(variable_id == "7" ~ "leaf_respiration_rate",
                              variable_id == "39" ~ "quantum_efficiency",
                              variable_id == "404" ~ "stomatal_slope.BB",
                              variable_id == "4" ~ "Vcmax",
                              variable_id == "2000000027" ~ "theta",
                              variable_id == "38" ~ "cuticular_cond"),
         site = case_when(site_id == "9000000005" ~ "Donald Danforth Plant Science Center Greenhouse",
                          site_id == "9000000004" ~ "Donald Danforth Plant Science Center Growth Chamber",
                          site_id == "9000000006" ~ "Donald Danforth Plant Science Center Outdoor"),
         SE = stat,
         entity = entity_id) %>%
  select(date, site, treatment, variable, mean, n, SE, entity) %>%
  arrange(site, treatment, variable, entity) %>%
  select(-date, -entity)
```

Create matching version of upload data
```{r}
upload <- in_upload %>%

>   select(site, treatment, variable, mean, n, SE, entity )%>%

Reviewer 2: ```suggestion
  select(site, treatment, variable, mean, n, SE, entity)%>%
```

  arrange(site, treatment, variable, entity) %>%
  select(-entity)
```

Compare both data versions
```{r}

> compare(bety, upload)

Reviewer 2: When I ran this, this is what is returned: 
```
Compare Object

Function Call: 
compare.data.frame(x = bety, y = upload)

Shared: 7 variables and 182 observations.
Not shared: 0 variables and 0 observations.

Differences found in 1/3 variables compared.
3 variables compared have non-identical attributes.
```
The two do look identical from some spot checking, what does the last bit mean about "non-identical attributes"? If that message is of no concern, it would be worthwhile to expand on that in the text below a bit.

Author: Thanks for pointing this out. There were some differences in character vs. factor, and a couple other things. Now all the correct lines read 0, but of course I'll have to update this after the add_SLA PR gets merged, haha!

Reviewer 2: This will probably have to continually be updated as data is added, including updating the Bety dump. It's neverending!
